client/etcd/client.go

```diff
@@ -244,7 +244,7 @@ func (c *csclient) etcdClientGen(zone string) (*clientv3.Client, error) {
 }
 
 func newClient(endpoints []string, certfile string, keyfile string, ca string) (*clientv3.Client, error) {
-	tlscfg := &tls.Config{}
+	var tlscfg *tls.Config
```

Comment:
While you're here, could you make the other changes we discussed in  #170

Reply:
Yeah would prefer to do it in next diff, I need to pull latest m3cluster in and this bug is blocking it.

Reply:
Cool, makes sense